src/analyses/base.ml

```diff
@@ -1105,7 +1105,7 @@ struct
       | Bot   -> Queries.ID.top () (* out-of-scope variables cause bot, but query result should then be unknown *)
       | Top   -> Queries.ID.top () (* some float computations cause top (57-float/01-base), but query result should then be unknown *)
       | v      -> M.debug ~category:Analyzer "Base EvalInt %a query answering bot instead of %a" d_exp e VD.pretty v; Queries.ID.bot ()
-      | exception (IntDomain.ArithmeticOnIntegerBot _) when not !AnalysisState.should_warn -> Queries.ID.top () (* for some privatizations, values can intermediately be bot because side-effects have not happened yet *)
+      | exception (IntDomain.ArithmeticOnIntegerBot _) -> Queries.ID.top ()
```

Comment:
This was introduced in #1049, but I wonder why it was conditional. Maybe the point was that it should not happen on the final solution (only on the described intermediate states), so if it did, then the exception should be propagated and reported to the user. Does that not happen here?

The other question is, why is there arithmetic on bottom values on the final solution at all?

Reply:
We thought a little bit on this and decided on re-establishing the bottom-free final solution arithmetics, but considered bottom to be the more appropriate value to propagate throughout eval_rv .
This is motivated due to EvalInt queries happening during recursive eval_rv calls; these queries might contribute insights from other analyses, like in our case apron. These insights might dominate the top value due to the meet during the query combination. Using bot instead, bot remains dominating other results, and is thus the more safe route for this case.

Reply:
@sim642 Does that sound reasonable to you?